enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,199 @@
+
+## Proposal
+
+Currently there is partial support provided by the `installer` (in UPI installations)
+to disable the `cluster-ingress-operator` from managing the DNS cluster-wide as documented
+[here](https://github.com/openshift/installer/blob/master/docs/user/aws/install_upi.md#remove-dns-zones). 
+
+The proposed solution adds support for more fine grained control over specific
+ingresscontrollers on how they manage the wildcard DNS records associated with them.
+
+### API Extensions
+
+The IngressController CRD is extended by adding an optional `ManageDNS`
+field with type `bool`. This field will default to true.
+
+```go
+// LoadBalancerStrategy holds parameters for a load balancer.
+type LoadBalancerStrategy struct {
+    // <snip>
+
+    // manageDNS indicates if the lifecyle of the wildcard DNS record
+    // associated with the load balancer service will be managed by
+    // the ingress operator.
+    //
+	// +kubebuilder:default:=true
+	// +kubebuilder:validation:Optional 
+    // +optional
+    ManageDNS bool `json:"manageDNS"`
+}
+```
+
+Re-uses existing state `DNSManaged` of the controller to denote to denote if
+the wildcard *DNSRecord* associated with the controller is managed/un-managed. 
+
+### Implementation Details/Notes/Constraints
+
+Based on `managedDNS`, the controller decides whether to ensure creating
+the a wildcard *DNSRecord*. If the customer is updating `managedDNS` from 
+`"true"` to `"false"` the previously created DNSRecord will be deleted.
+
+The support provided by the installer and this new field aren't entriely
+mutually exclusive i.e. if customer has opted to disable DNS management cluster
+wide (as mentioned under UPI based installations), this additional flag is of
+no value and need not be set since *DNSRecord* created is silently reconciled 
+by the dns controller.
+
+### Risks and Mitigations
+
+The manual creation of the required DNS records will need to be done prior to
+making this change as the canary controller will try to ensure the if the ingress
+domains are reachable, otherwise the `ingress` *ClusterOperator* results in a 
+degraded state. This needs to be documented so as to avoid unecessary support cases.
+
+### Drawbacks
+
+There is an inconsistency in how the existing support is added to disable
+DNS management cluster-wide by the ingress controller i.e. the controller 
+proceedes to create a wildcard *DNSRecord* and during reconcilliation the dns
+controller [silently skips](https://github.com/openshift/cluster-ingress-operator/blob/972f09b9dbb181ae5c414da2a990b57c60fde9d8/pkg/operator/controller/dns/controller.go#L261-L311)
+publishing the record to the zone. Where as in the proposed solution, 
+a wildcard *DNSRecord* is **not** created when `manageDNS` is set to `false`. 
+
+## Design Details
+
+### Test Plan
+
+- Test by updating the default ingresscontroller to be associated with a manually
+  created dns record and ensure correct working of test workload.
```

Comment:
I imagine that the cluster provisioned by the CI will have the cluster DNS zone and the cluster ingress operator installed, so the wildcard for the routes will already be there when this test case will hit. This means that even if you setup the wildcard DNS somewhere else and disable the dns management for the default ingress controller: how would you make sure that the DNS resolution is still not going from the old DNS records (created at the installation)?

Reply:
Within the test case, we would be updating the `ingresscontroller` to update the `domain` and `manageDNS` field to the required values. Followed by creating a new DNS zone and record which is associated to the default *LoadBalancerService* `router-default`'s external IP. 

The update will ensure the older *DNSRecord* is deleted and using something like an echo server (workload) the new domain will be tested.